data/prepare_data.py

```diff
@@ -70,8 +86,45 @@ def run(self) -> (DatasetDict, Dict):
                 token_ids, attention_masks, token_type_ids, labels = prepare_input_rel_cls(
                     tokenized_sentences, labels_mapper, categor_labels, self.device
                 )
-
+            elif self.task == "dep":
+                token_ids_dep, attention_masks_dep, token_type_ids_dep, labels_dep = prepare_input_dep(tokenized_sentences_dep, 
+                                                                                                       self.tokenizer, 
+                                                                                                       labels_mapper_dep,
+                                                                                                       self.device)                      
+                
+                token_ids_heads, attention_masks_heads, token_type_ids_heads, labels_heads = prepare_input_dep(tokenized_sentences_heads, 
+                                                                                                               self.tokenizer, 
+                                                                                                               labels_mapper_head,
+                                                                                                               self.device)
             # Create a Dataset object
+
+            directory = path.split(os.path.sep)
+            dataset_name = directory[-1]
+
+            if self.task == "dep":
+                dataset_inputs_dep = Dataset.from_dict(
+                    {
+                        "input_ids": token_ids_dep,
+                        "attention_mask": attention_masks_dep,
+                        "token_type_ids": token_type_ids_dep,
+                        "labels": labels_dep,
+                    }
+                )
+                dataset_inputs_heads = Dataset.from_dict(
+                    {
+                        "input_ids": token_ids_heads,
+                        "attention_mask": attention_masks_heads,
+                        "token_type_ids": token_type_ids_heads,
+                        "labels": labels_heads,
+                    }
+                )
+
+                dataset_dict["dep"] = dataset_inputs_dep
+                dataset_dict["heads"] = dataset_inputs_heads
+
+                dataset_dict[dataset_name[:-4]] = {"dep": dataset_inputs_dep, "heads": dataset_inputs_heads}
+
+
             dataset_inputs = Dataset.from_dict(
```

Comment:
Lines from here forward should be executed only if the task is not dep, so they should be in an else statement, I think. 
Alternatively, the second if statement in line 139 can be moved to be inside this if.